akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -662,23 +661,6 @@ private[akka] class Shard(
       }
     case RestartRememberedEntities(entities) => restartEntities(entities)
     case l: LeaseLost                        => receiveLeaseLost(l)
-    case ShardRegion.StartEntityAck(entityId, _) =>
-      if (update.started.contains(entityId)) {
-        // currently in progress of starting, so we'll need to stop it when that is done
-        entities.rememberingStop(entityId, StartedElsewhere)
-      } else {
-        entities.entityState(entityId) match {
-          case _: RememberingStart =>
-            // queued up for batched start, let's just not start it
-            entities.removeEntity(entityId)
```

Comment:
This is perhaps what we lose, cleaning up entities in the `RememberedButNotCreated` state in `entities`

Reply:
We could cover that by going through the remembered entities when the RememberEntitiesStarter completes and prune instead?